mqt/qecc/bindings.cpp

```
@@ -24,7 +35,77 @@ std::vector<bool> sampleIidPauliErr(const std::size_t length, const double physi
+            py::object const pyQasmQobjExperiment = py::module::import("qiskit.qobj").attr("QasmQobjExperiment");
+            if (py::isinstance(circ, quantumCircuit)) {
+                qc::qiskit::QuantumCircuit::import(*qc, circ);
+            } else if (py::isinstance(circ, pyQasmQobjExperiment)) {
+                qc::qiskit::QasmQobjExperiment::import(*qc, circ);
+            }
+        }
+    } catch (std::exception const& e) {
+        std::stringstream ss{};
+        ss << "Could not import circuit: " << e.what();
+        return py::dict("error"_a = ss.str());
+    }
+
+    auto test = std::make_unique<ecc::Id>(qc, eccFrequency);
+
+    [[maybe_unused]] std::unique_ptr<ecc::Ecc> mapper{};
+
+    if (eccName == "Id") {
+        mapper = std::make_unique<ecc::Id>(qc, eccFrequency);
+    } else if (eccName == "Q3Shor") {
+        mapper = std::make_unique<ecc::Q3Shor>(qc, eccFrequency);
+    } else if ((eccName == "Q5Laflamme")) {
+        mapper = std::make_unique<ecc::Q5Laflamme>(qc, eccFrequency);
+    } else if ((eccName == "Q7Steane")) {
+        mapper = std::make_unique<ecc::Q7Steane>(qc, eccFrequency);
+    } else if ((eccName == "Q9Shor")) {
+        mapper = std::make_unique<ecc::Q9Shor>(qc, eccFrequency);
+    } else if ((eccName == "Q9Surface")) {
+        mapper = std::make_unique<ecc::Q9Surface>(qc, eccFrequency);
+    } else if ((eccName == "Q18Surface")) {
+        mapper = std::make_unique<ecc::Q18Surface>(qc, eccFrequency);
+    } else {
+        std::stringstream ss{};
+        ss << "No ECC found for " << eccName << " ";
+        ss << "Available ECCs: ";
+        ss << "Id"
+           << ", ";
+        ss << "Q3Shor"
+           << ", ";
+        ss << "Q5Laflamme"
+           << ", ";
+        ss << "Q7Steane"
+           << ", ";
+        ss << "Q9Shor"
+           << ", ";
+        ss << "Q9Surface"
+           << ", ";
+        ss << "Q18Surface";
+        return py::dict("error"_a = ss.str());
```

Instead of returning an error message, just throw an appropriate exception (e.g. "std::invalid_argument") here. This automatically gets converted to a proper python exception when triggered.

---

Throwing an exception instead.